src/chips/sam4l/lib.rs

```diff
@@ -129,24 +146,24 @@ pub static ISR_VECTOR: [Option<unsafe extern fn()>; 96] = [
     /* TC10 */          Option::Some(unhandled_interrupt),
     /* TC11 */          Option::Some(unhandled_interrupt),
     /* TC12 */          Option::Some(unhandled_interrupt),
-    /* TWIM0 */         Option::Some(unhandled_interrupt),
+    /* TWIM0 */         Option::Some(i2c::twim0_handler),
     /* TWIS0 */         Option::Some(unhandled_interrupt),
-    /* TWIM1 */         Option::Some(unhandled_interrupt),
+    /* TWIM1 */         Option::Some(i2c::twim1_handler),
     /* TWIS1 */         Option::Some(unhandled_interrupt),
-    /* USART0 */        Option::Some(unhandled_interrupt),
-    /* USART1 */        Option::Some(unhandled_interrupt),
-    /* USART2 */        Option::Some(usart::USART2_Handler),
-    /* USART3 */        Option::Some(usart::USART3_Handler),
-    /* ADCIFE */        Option::Some(adc::ADCIFE_Handler),
+    /* USART0 */        Option::Some(usart::usart0_handler),
+    /* USART1 */        Option::Some(usart::usart1_handler),
+    /* USART2 */        Option::Some(usart::usart2_handler),
+    /* USART3 */        Option::Some(usart::usart3_handler),
+    /* ADCIFE */        Option::Some(adc::adcife_handler),
     /* DACC */          Option::Some(unhandled_interrupt),
     /* ACIFC */         Option::Some(unhandled_interrupt),
     /* ABDACB */        Option::Some(unhandled_interrupt),
     /* TRNG */          Option::Some(unhandled_interrupt),
     /* PARC */          Option::Some(unhandled_interrupt),
     /* CATB */          Option::Some(unhandled_interrupt),
     None,
```

Comment:
Hrm.... Good point. Probably don't need them. It would be nice if there was a way to use an enum equivilant, where the `None` version points to `unhandled_interrupt` -- I think that would justify something like this, but I don't think this is possible without yet another level of indirection or deeper integration in the assembly